File: testdata/testkit/lua/src/entity.lua
local socket = require("socket")

local Entity = {}
Entity.__index = Entity

function Entity.new(id, name)
    return setmetatable({ id = id, name = name, data = {} }, Entity)
end

function Entity:toString()
    return string.format("Entity(%s, %s)", self.id, self.name)
end

-- Repository via metatable (Strategy pattern)
local Repository = {}
Repository.__index = Repository

function Repository.new()
    return setmetatable({ store = {} }, Repository)
end

function Repository:findById(id)
    return self.store[id]
end

function Repository:save(entity)
    self.store[entity.id] = entity
end

return { Entity = Entity, Repository = Repository }

File: testdata/testkit/cpp/src/entity.cpp
#include "../include/entity.hpp"
#include <unordered_map>

class InMemoryRepo : public Repository {
    std::unordered_map<std::string, Entity> store_;
public:
    std::optional<Entity> find_by_id(const std::string& id) override {
        auto it = store_.find(id);
        if (it == store_.end()) return std::nullopt;
        return it->second;
    }
    void save(const Entity& entity) override {
        store_.emplace(entity.id(), entity);
    }
};
